src/System.Private.CoreLib/shared/System/Runtime/Serialization/SerializationInfo.cs

```diff
@@ -23,6 +26,137 @@ public sealed class SerializationInfo
         private string _rootTypeName;
         private string _rootTypeAssemblyName;
         private Type _rootType;
+        
+        internal static AsyncLocal<bool> AsyncDeserializationInProgress { get; } = new AsyncLocal<bool>();        
```

Comment:
Nit: readonly?

Reply:
Nit: s_asyncDeserializationInProgress?

Reply:
It's currently a get-only property, not a static field.

Reply:
Ah, yup